### PR TITLE
Create base model

### PR DESCRIPTION
**Pull request by minzcmu**
- Create base model 
- Base model has get, list and update methods for re-use in build, job and pipeline

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 'use strict';
 const Build = require('./lib/buildmodel');
+const Job = require('./lib/jobmodel');
```
Comment (stjohnjohnson) [inline index.js]: Can we please use a better file naming schema here.

Something like:
- `./lib/models/job`
- `./lib/job`

File: lib/basemodel.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+class BaseModel {
+
```
Comment (stjohnjohnson) [inline lib/basemodel.js]: Extra line

**Comment by stjohnjohnson** [comment]
LGTM (besides the file naming, which can be addressed in another PR if you want)

File: index.js (hunk 1 above)
Comment (d2lam) [inline index.js]: ok we will change it to `./lib/job` since it's given that these are models.

**Comment by tkyi** [comment]
👍 LGTM

**Pull request merged**